pkg/controller/awsfederatedrole/awsfederatedrole_controller.go

```diff
@@ -135,45 +136,45 @@ func (r *ReconcileAWSFederatedRole) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{}, nil
 	}
 
-	// Attemps to create the policy to ensure its a valid policy
-	createOutput, err := awsClient.CreatePolicy(&iam.CreatePolicyInput{
-		Description:    &instance.Spec.AWSCustomPolicy.Description,
-		PolicyName:     &instance.Spec.AWSCustomPolicy.Name,
-		PolicyDocument: &jsonPolicy,
-	})
+	// Get aws id and create policy arn
+	gciOut, err := awsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			if aerr.Code() == "MalformedPolicyDocument" {
-				log.Error(err, "Malformed Policy Document")
-				instance.Status.State = awsv1alpha1.AWSFederatedRoleStateInvalid
-				instance.Status.Conditions = utils.SetAWSFederatedRoleCondition(
-					instance.Status.Conditions,
-					awsv1alpha1.AWSFederatedRoleInvalid,
-					"True",
-					"InvalidCustomerPolicy",
-					"Custom Policy is malformed",
-					utils.UpdateConditionNever)
-				err = r.client.Status().Update(context.TODO(), instance)
-				if err != nil {
-					log.Error(err, "Error updating conditions")
-					return reconcile.Result{}, err
-				}
-				return reconcile.Result{}, nil
-			}
-			utils.LogAwsError(log, "", nil, err)
-		} else {
-			log.Error(err, "Non-AWS Error while creating Policy")
-		}
+		reqLogger.Error(err, fmt.Sprintf("Unable to create account id requested"))
 		return reconcile.Result{}, err
 	}
 
+	// policy arn template
+	policyARN := fmt.Sprintf("arn:aws:iam::%s:policy/%s", *gciOut.Account, instance.Spec.AWSCustomPolicy.Name)
+
+	// Validates Custom IAM Policy
+	log.Info("Validating Custom Policies")
+	err = awsfaa.CreateOrUpdateIAMPolicy(awsClient, *instance, policyARN)
+	// Invalidates instance if unable not create policy
+	if err != nil {
```

Comment:
Why do we no longer handle the AWS error? We need to unwrap it and log it somewhere like every other AWS error in the operator